search/includes/classes/class-search.php

```diff
@@ -237,4 +240,14 @@ static function ep_skip_query_integration( $skip ) {
 		// The filter is checking if we should _skip_ query integration
 		return ! ( $query_integration_enabled || $query_integration_enabled_legacy );
 	}
+
+	public function filter__ep_valid_response( $response, $query, $query_args, $query_object ) {
+		if ( ! headers_sent() ) {
+			/**
+			 * Manually set a header to indicate the search results are from Elasticsearch
+			 */
+			header( 'X-ElasticPress-Search-Valid-Response: true' );
```

Comment:
Should we name this to be more specific to VIP Search? Also ElasticPress already sends a header which means we now have two headers with this change?

Reply:
ElasticPress sends a header when it's involved in the process in any capacity, even if the query never hits ES. 

The idea of this header is to give an indication of when ES is actually involved. See [this Slack thread](https://a8c.slack.com/archives/C03CU7X3U/p1583338418088500) for context.

Reply:
Yes, there are two headers, but we mainly want to know "did the search results actually come from ElasticPress"
The baked in header seems kind of.. less useful, but I don't think we want to swap it out or remove it.

Reply:
If the original EP header is not very useful, we should definitely consider either removing it or making it more useful. (Although, I guess it may be impossible to do if a EP query/request is triggered halfway through a page-render).

My main concern is additional unnecessary bytes we're adding to each pageload. At a micro-level it's not a big deal, but globally, it adds up to a lot :)